SimTracker/TrackerMaterialAnalysis/plugins/TrackingMaterialProducer.cc

```diff
@@ -281,12 +283,24 @@ void TrackingMaterialProducer::update(const G4Step* step) {
   }
 
   // update track accounting
-  if (enter_sensitive)
-    m_track.enterDetector(sensitive, position, cosThetaPre);
+  if (enter_sensitive) {
+    if (m_track_volume != nullptr) {
+      edm::LogWarning("TrackingMaterialProducer") << "Entering volume " << sensitive << "while inside volume "
+                                                  << m_track_volume << ". Something is inconsistent";
```

Comment:
Hello @davidlange6. If this is inconsistent, is a warning enough?

Reply:
No idea. it was a cerr which I wanted to make a message logger

Reply:
@davidlange6 by running:

`cmsRun SimTracker/TrackerMaterialAnalysis/test/trackingMaterialProducer10GeVNeutrino_ForPhaseI.py`

Now I get thousands of printouts like this:

```
Entering volume TECModule2RphiActivewhile inside volume TECModule1StereoActive. Something is inconsistent
Entering volume TECModule1RphiActivewhile inside volume TECModule2RphiActive. Something is inconsistent
Entering volume TECModule1StereoActivewhile inside volume TECModule1RphiActive. Something is inconsistent
Entering volume TECModule2RphiActivewhile inside volume TECModule1StereoActive. Something is inconsistent
Entering volume TECModule2RphiActivewhile inside volume TECModule2RphiActive. Something is inconsistent
Entering volume TECModule3RphiActivewhile inside volume TECModule2RphiActive. Something is inconsistent
Entering volume TECModule3RphiActivewhile inside volume TECModule3RphiActive. Something is inconsistent
Entering volume TECModule4StereoActivewhile inside volume TECModule3RphiActive. Something is inconsistent
Entering volume TECModule4RphiActivewhile inside volume TECModule4StereoActive. Something is inconsistent
Entering volume TECModule3RphiActivewhile inside volume TECModule4RphiActive. Something is inconsistent
Entering volume TECModule4StereoActivewhile inside volume TECModule3RphiActive. Something is inconsistent
Entering volume TECModule4RphiActivewhile inside volume TECModule4StereoActive. Something is inconsistent
Entering volume PixelBarrelActiveFull1while inside volume PixelBarrelActiveFull0. Something is inconsistent
Entering volume PixelBarrelActiveFull2while inside volume PixelBarrelActiveFull1. Something is inconsistent
Entering volume PixelBarrelActiveFull3while inside volume PixelBarrelActiveFull2. Something is inconsistent
Entering volume TIBActiveRphi0while inside volume PixelBarrelActiveFull3. Something is inconsistent
Entering volume TIBActiveSter0while inside volume TIBActiveRphi0. Something is inconsistent
Entering volume TIBActiveRphi0while inside volume TIBActiveSter0. Something is inconsistent
Entering volume TIBActiveSter0while inside volume TIBActiveRphi0. Something is inconsistent
Entering volume TIBActiveRphi2while inside volume TIBActiveSter0. Something is inconsistent
Entering volume TIBActiveRphi2while inside volume TIBActiveRphi2. Something is inconsistent
Entering volume TOBActiveRphi0while inside volume TIBActiveRphi2. Something is inconsistent
Entering volume TOBActiveRphi0while inside volume TOBActiveRphi0. Something is inconsistent
Entering volume TOBActiveSter0while inside volume TOBActiveRphi0. Something is inconsistent
Entering volume TOBActiveRphi0while inside volume TOBActiveSter0. Something is inconsistent
Entering volume TOBActiveRphi2while inside volume TOBActiveRphi0. Something is inconsistent
Entering volume TOBActiveRphi2while inside volume TOBActiveRphi2. Something is inconsistent
Entering volume PixelForwardSensorwhile inside volume PixelBarrelActiveFull0. Something is inconsistent
Entering volume PixelForwardSensorwhile inside volume PixelForwardSensor. Something is inconsistent
Entering volume PixelForwardSensorwhile inside volume PixelForwardSensor. Something is inconsistent
Entering volume PixelForwardSensorwhile inside volume PixelForwardSensor. Something is inconsistent
Entering volume PixelBarrelActiveFull1while inside volume PixelBarrelActiveFull0. Something is inconsistent
Entering volume PixelBarrelActiveFull2while inside volume PixelBarrelActiveFull1. Something is inconsistent
```
Can you please double check that these changes do not change the expected behavior? as I pointed out before merging histograms from the unit test are also empty. Thanks

Reply:
Fixed in #31752